lib/ansible/modules/cloud/memset/memset_memstore_user.py

```diff
@@ -0,0 +1,337 @@
+    retvals, payload = dict(), dict()
+    retvals['changed'], retvals['failed'] = False, False
+
+    if user:
+        if args['check_mode']:
+            retvals['changed'] = True
+            return(retvals)
+        payload['name'] = args['memstore']
+        payload['username'] = args['username']
+        api_method = 'memstore.user.delete'
+        retvals['failed'], msg, response = memset_api_call(api_key=args['api_key'], api_method=api_method, payload=payload)
+
+        if retvals['failed']:
+            retvals['msg'] = msg
+        else:
+            retvals['changed'] = True
+
+    return(retvals)
+
+
+def update_password(args=None):
+    '''
+    Updates the password of an existing user.
+    '''
+    retvals, payload = dict(), dict()
+    retvals['changed'], retvals['failed'] = False, False
+
+    if args['check_mode']:
+        # Updating a password will always return changed as the API does not
+        # know if the password is actually being changed.
+        retvals['password'] = 'Check mode cannot tell if the password will be changed.'
+    else:
+        payload['name'], payload['username'], payload['password'] = args['memstore'], \
+            args['username'], args['password']
+
+        api_method = 'memstore.user.set_password'
+        retvals['failed'], msg, response = memset_api_call(args['api_key'], api_method, payload)
+
+        if retvals['failed']:
+            retvals['password'] = msg
+        else:
+            retvals['changed'] = True
+            retvals['password'] = 'Password updated.'
+
+    return(retvals)
+
+
+def create_or_delete_user(args=None):
+    '''
+    Performs initial auth validation and gets a list of
+    existing services to provide to create/delete functions.
+    '''
+    retvals, payload = dict(), dict()
+    retvals['changed'], retvals['failed'] = False, False
+
+    payload['name'] = args['memstore']
+    api_method = 'memstore.user.list'
+    has_failed, msg, response = memset_api_call(api_key=args['api_key'], api_method=api_method, payload=payload)
+
+    if has_failed:
+        # this is the first time the API is called; incorrect credentials will
+        # manifest themselves at this point so we need to ensure the user is
+        # informed of the reason.
+        retvals['failed'] = has_failed
+        retvals['msg'] = msg
+        return(retvals)
+
+    _user = None
+    for user in response.json():
+        if user['username'] == args['username']:
+            _user = user
+            break
+
+    if args['state'] == 'present':
+        # the user may already exist, however we may also want to update
+        # their attributes.
+        retvals = create_user(args=args, user=_user)
+    if args['state'] == 'absent':
+        # remove user
+        retvals = delete_user(args=args, user=_user)
+
+    return(retvals)
+
+
+def main():
+    global module
+
+    required_if = [
+        ['state', 'present', ['password']],
+        ['update_password', 'always', ['password']]
```

Comment:
I don't think you need this line: `password` (and `update_password`) are only needed when `state` is `present`, and in that case, `password` is already required.

Reply:
(Alternatively, you could remove the line above, and let the module fail if the user does not exist, `state` is `present` and the password not specified.)

Reply:
I think I prefer the first option - just pushed an update to remove the line entirely. I think I'd rather just restrict the user than let them do something which won't work.